### PR TITLE
Add docs for testing export of configuration

### PR DESCRIPTION
**Pull request by timja**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [ ] Link to relevant GitHub issues or pull requests

- [ ] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

Update plugin developers guide for exporting configuration
Based on the code here:
https://github.com/jenkinsci/azure-keyvault-plugin/blob/master/src/test/java/org/jenkinsci/plugins/azurekeyvaultplugin/ConfigAsCodeTest.java#L33

### CODE REVIEW

File: docs/PLUGINS.md [hunk 1]
```diff
@@ -277,9 +277,18 @@ You also can write a test case to check export from a live instance is well supp
...
 
-**TODO** we need to provide some YAML assertion library so that the resulting exported yam stream can be checked for expected content.
+        @SuppressWarnings("unchecked")
+        CNode node = c.describe(globalConfiguration, context);
+        assertNotNull(node);
+        final Mapping mapping = node.asMapping();
+
+        assertEquals(mapping.getScalarValue("expected-property"), "expected-value");
+        assertEquals(mapping.getScalarValue("another-property"), "another-value");
+
```
Comment (casz) [inline docs/PLUGINS.md]: Perhaps another example of getting a value from a `Sequence`

Comment (timja) [reply docs/PLUGINS.md]: added based on matrix-auth changes

File: docs/PLUGINS.md [hunk 2]
```diff
@@ -290,5 +290,8 @@ You also can write a test case to check export from a live instance is well supp
 
         assertEquals(mapping.getScalarValue("expected-property"), "expected-value");
         assertEquals(mapping.getScalarValue("another-property"), "another-value");
+        
+        List<CNode> yourList = mapping.get("a-list-property").asSequence();
+        assertEquals("list size", 5, yourList.size());
```
Comment (casz) [inline docs/PLUGINS.md]: `yourList.get(0).getScalar().getValue()`

`yourList.get(0).asMapping().getScalarValue('key')`

Comment (casz) [reply docs/PLUGINS.md]: actually I'd like for `CNode` to have the same shortcuts `.get(0).getScalarValue()` and `.get(0).getScalarValue('key')` which handles the scalar and mapping conversion

**Pull request merged**